### PR TITLE
DDCNL-3127: Added dummydata overwrite

### PR DESCRIPTION
**Pull request by danewatts**

### CODE REVIEW

File: test/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/SandboxTestDataControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+package uk.gov.hmrc.apprenticeshiplevy.controllers.sandbox
+
+import org.scalatestplus.play.PlaySpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Configuration
+import uk.gov.hmrc.apprenticeshiplevy.controllers.sandbox.data.SandboxTestDataController
+
+class SandboxTestDataControllerSpec extends PlaySpec with GuiceOneAppPerSuite {
```
Comment (liampan) [inline test/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/SandboxTestDataControllerSpec.scala:24]: do you need GuiceOneAppPerSuite?

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala [hunk 2]
```diff
@@ -19,40 +19,78 @@ package uk.gov.hmrc.apprenticeshiplevy.controllers.sandbox.data
...
+    configuration
+      .getBoolean("features.returnDummyResponse")
+      .getOrElse(false)
+
+  private[sandbox] def getFileName(file: String) =
+    if(returnDummyResponse) {
+      val empRefRegex = "^.*employers?\\/([a-zA-Z0-9]{3}\\/.*?|.*?)\\/.*".r
+      file match {
+        case empRefRegex(empRef) =>
+          Logger.debug(s"Replacing empref: $empRef in requested file: $file")
```
Comment (liampan) [inline app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala:49]: do we want a class logger instead?

File: test/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/SandboxTestDataControllerSpec.scala (hunk 1 above)
Comment (danewatts) [reply test/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/SandboxTestDataControllerSpec.scala:24]: No - I added this in and moved away from loading the App. Ill remove it

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala (hunk 2 above)
Comment (danewatts) [reply app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala:49]: I can make a custom logger instance

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala [hunk 3]
```diff
@@ -19,40 +19,78 @@ package uk.gov.hmrc.apprenticeshiplevy.controllers.sandbox.data
...
+@Singleton
+class SandboxTestDataController @Inject()(configuration: Configuration) extends Controller with Utf8MimeTypes {
   val SANDBOX_DATA_DIR = "public/sandbox-data"
 
+  lazy val returnDummyResponse: Boolean =
+    configuration
+      .getBoolean("features.returnDummyResponse")
+      .getOrElse(false)
+
+  private[sandbox] def getFileName(file: String) =
```
Comment (lewey) [inline app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala:44]: missing return type

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala (hunk 2 above)
Comment (lewey) [inline app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala:49]: class logger

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala [hunk 4]
```diff
@@ -19,40 +19,78 @@ package uk.gov.hmrc.apprenticeshiplevy.controllers.sandbox.data
...
+    if(returnDummyResponse) {
+      val empRefRegex = "^.*employers?\\/([a-zA-Z0-9]{3}\\/.*?|.*?)\\/.*".r
+      file match {
+        case empRefRegex(empRef) =>
+          Logger.debug(s"Replacing empref: $empRef in requested file: $file")
+          val dummyEmpRef = List("840","MODES17")
+          val ninoRegex = "^.*\\/employed\\/(.{8,9})\\?.*$".r
+
+          val updatedFile = file.replace(empRef,
```
Comment (lewey) [inline app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala:53]: Nested if could be read easier if it were a val. Preference I guess though.

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala (hunk 3 above)
Comment (danewatts) [reply app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala]: Added

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala (hunk 2 above)
Comment (danewatts) [reply app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala]: Have done

File: app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala (hunk 4 above)
Comment (danewatts) [reply app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala]: I have pulled the if into a val

**Review by liampan — APPROVED** [review]

**Pull request merged**